dpnp/backend/kernels/dpnp_krnl_fft.cpp

```diff
@@ -215,98 +221,184 @@ void dpnp_fft_fft_mathlib_compute_c(DPCTLSyclQueueRef q_ref,
+void dpnp_fft_fft_mathlib_real_to_cmplx_c(DPCTLSyclQueueRef q_ref,
+                                          const void* array1_in,
+                                          void* result_out,
+                                          const shape_elem_type* input_shape,
+                                          const shape_elem_type* result_shape,
+                                          const size_t shape_size,
+                                          const size_t input_size,
+                                          const size_t result_size,
+                                          _Descriptor_type& desc,
+                                          const size_t norm,
+                                          const size_t real)
 {
-    if (!shape_size || !result_size || !array1_in || !result1)
+    if (!shape_size)
     {
         return;
     }
-    //will be used with strides
-    //std::vector<std::int64_t> dimensions(input_shape, input_shape + shape_size);
 
-    if constexpr (std::is_same<_DataType_input, std::complex<double>>::value &&
-                  std::is_same<_DataType_output, std::complex<double>>::value)
-    {
-        desc_dp_cmplx_t desc(input_shape[shape_size - 1]);
+    DPNPC_ptr_adapter<_DataType_input> input1_ptr(q_ref, array1_in, input_size);
+    DPNPC_ptr_adapter<_DataType_output> result_ptr(q_ref, result_out, result_size * 2, true, true);
+    _DataType_input* array_1 = input1_ptr.get_ptr();
+    _DataType_output* result = result_ptr.get_ptr();
+
+    sycl::queue queue = *(reinterpret_cast<sycl::queue*>(q_ref));
+
+    const size_t n_iter =
+        std::accumulate(input_shape, input_shape + shape_size - 1, 1, std::multiplies<shape_elem_type>());
+
+    const size_t input_shift = input_shape[shape_size - 1];
+    const size_t result_shift = result_shape[shape_size - 1];;
 
-        dpnp_fft_fft_mathlib_compute_c<_DataType_input, _DataType_output, desc_dp_cmplx_t>(
-            q_ref, array1_in, result1, input_shape, shape_size, result_size, desc, norm);
+    double forward_scale = 1.0;
+    double backward_scale = 1.0 / input_shift;
+
+    desc.set_value(mkl_dft::config_param::BACKWARD_SCALE, backward_scale);
+    desc.set_value(mkl_dft::config_param::FORWARD_SCALE, forward_scale);
+
+    desc.commit(queue);
+
+    std::vector<sycl::event> fft_events;
+    fft_events.reserve(n_iter);
+
+    for (size_t i = 0; i < n_iter; ++i) {
+        fft_events.push_back(mkl_dft::compute_forward(desc, array_1 + i * input_shift, result + i * result_shift * 2));
     }
-    else if (std::is_same<_DataType_input, std::complex<float>>::value &&
-             std::is_same<_DataType_output, std::complex<float>>::value)
-    {
-        desc_sp_cmplx_t desc(input_shape[shape_size - 1]);
 
-        dpnp_fft_fft_mathlib_compute_c<_DataType_input, _DataType_output, desc_sp_cmplx_t>(
-            q_ref, array1_in, result1, input_shape, shape_size, result_size, desc, norm);
+    sycl::event::wait(fft_events);
+
+    if (!real) {
```

Comment:
Could we return from function if real is 1 and add comment why? Then shift else-code to the left.